### DISCUSSION TITLE
How to add source file?

### DISCUSSION THREAD

**Discussion by mattjes** [silgy/nodepp#15] [Q&A] [answered]
Hello I'm a noob. This package is exactly what I want. Only problem is I am a noob and I am not used to compiling a program in the way this is compiled. 

Could someone point me in the right direction on how to add extra src files to the compile job?

**Comment by rekmus** [comment, marked as answer]
Hi Mattjes!

This is actually a very good question and I just realized it's not described properly in the documentation.

In **src/npp_app.h** there's **NPP_APP_MODULES** where you can add your additional source files like this:

`
#define NPP_APP_MODULES "additional1.cpp additional2.c"
`

That's it. This file is parsed by the compilation script so you don't have to add them anywhere else.

**Reply by mattjes** [reply]
Hi thanks for your reply.

I tried it and I works 👍 

Additionally I would like to use the pthread library which I have done before using -pthread. If done in the following way it seems to compile with the library properly.

`#define NPP_APP_MODULES "-pthread additional.cpp"`

Thanks for making this it looks great, am running it on a raspberry pi.

**Reply by rekmus** [reply]
I appreciate your feedback, thanks!
